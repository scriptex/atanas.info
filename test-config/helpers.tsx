import { act } from 'react-dom/test-utils';
import type { FC } from 'react';
import { render, RenderResult, waitFor } from '@testing-library/react';

import * as shared from '@scripts/shared';

type ComponentType = FC<Readonly<React.ComponentProps<any>>>;

export const test = async (Component: ComponentType): Promise<RenderResult> => {
	let result!: RenderResult;

	await act(async () => {
		result = await waitFor(() => render(<Component />));
	});

	return result;
};

export const snapshotTest = (Component: ComponentType, buttonSelector?: string): void => {
	const { name } = Component;

	jest.spyOn(shared, 'formatDate');

	(shared.formatDate as jest.Mock).mockImplementation(() => 'Mock date');

	afterAll(() => {
		jest.clearAllMocks();
	});

	describe(`${name} component`, () => {
		it(`Should render the ${name} component`, async () => {
			const { asFragment, container } = await test(Component);

			expect(asFragment()).toMatchSnapshot();

			if (buttonSelector) {
				await act(async () => {
					await container.querySelectorAll<HTMLElement>(buttonSelector).forEach(button => {
						if (!button.hasAttribute('download')) {
							button.click();

							expect(asFragment()).toMatchSnapshot();
						}
					});
				});
			}
		});

		it(`Should unmount the ${name} component`, async () => {
			const { unmount, asFragment } = await test(Component);

			unmount();

			expect(asFragment()).toMatchSnapshot();
		});
	});
};

export const mockAudioContext = (): jest.Mock => {
	const mockConnect = jest.fn();

	const mockgetByteFrequencyData = jest.fn();

	const mockcreateMediaElementSource = jest.fn(() => ({
		connect: mockConnect
	}));

	const mockcreateAnalyser = jest.fn(() => ({
		connect: mockConnect,
		frequencyBinCount: [0, 1, 2],
		getByteFrequencyData: mockgetByteFrequencyData
	}));

	const mockcreateOscillator = jest.fn(() => ({
		channelCount: 2
	}));

	const mockChannelSplitterConnect = jest.fn(n => n);

	const mockcreateChannelSplitter = jest.fn(() => ({
		connect: mockChannelSplitterConnect
	}));

	const mockedAudioContext = jest.fn(() => ({
		createAnalyser: mockcreateAnalyser,
		createOscillator: mockcreateOscillator,
		createChannelSplitter: mockcreateChannelSplitter,
		createMediaElementSource: mockcreateMediaElementSource
	}));

	return mockedAudioContext;
};

export const mockFetch = (value: Record<string, string | boolean>): void => {
	global.fetch = jest.fn().mockImplementation(() =>
		Promise.resolve({
			json: () => Promise.resolve(value)
		})
	);
};
