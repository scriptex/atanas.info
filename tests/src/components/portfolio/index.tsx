import { snapshotTest } from '@test-config/helpers';
import { PortfolioSliders } from '@components';
import { automotiveProjects } from '@data/projects';

snapshotTest(
	() => (
		<PortfolioSliders
			data={automotiveProjects.map((item, index) => (index === 0 ? { ...item, adjustable: true } : item))}
			folder="automotive-apps"
			slidesToShow={2}
			className="test"
		/>
	),
	undefined,
	'PortfolioSliders'
);
