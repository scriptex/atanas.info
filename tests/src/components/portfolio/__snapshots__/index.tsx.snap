// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unknown component Should render the Unknown component 1`] = `
<DocumentFragment>
  <div
    class="c-section__sliders"
  >
    <div
      class="c-section__slider test"
    >
      <div
        class="c-loading"
      >
        <div
          class="c-loader"
        />
      </div>
      <a
        href="https://kinetik-infotainment.netlify.app/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h4>
          Kinetik Infotainment
        </h4>
        <p>
          Dashboard and infotainment system for the Kinetik Model 27 electric kart
        </p>
        <code>
          Technologies used:
          <br />
          Typescript, SCSS, SVG, React, Redux, NodeJS, Express, WebSocket
        </code>
        <svg
          class="c-svg-external-link"
        >
          <use
            xlink:href="#svg-external-link"
          />
        </svg>
      </a>
      <div
        class="slick-slider slick-initialized"
        dir="ltr"
      >
        <button
          class="slick-arrow slick-prev"
          data-role="none"
          style="display: block;"
          type="button"
        >
           Previous
        </button>
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="-2"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 8"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-8.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-8.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="-1"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 9"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-9.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-9.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 1"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-1.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-1.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 2"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-2.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-2.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="2"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 3"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-3.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-3.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="3"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 4"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-4.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-4.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="4"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 5"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-5.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-5.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="5"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 6"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-6.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-6.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="6"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 7"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-7.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-7.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="7"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 8"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-8.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-8.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="8"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 9"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-9.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-9.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="9"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 1"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-1.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-1.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="10"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 2"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-2.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-2.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="11"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 3"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-3.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-3.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="12"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 4"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-4.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-4.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="13"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 5"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-5.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-5.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="14"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 6"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-6.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-6.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="15"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 7"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-7.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-7.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="16"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 8"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-8.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-8.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="17"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Kinetik Infotainment screenshot 9"
                    data-nimg="1"
                    decoding="async"
                    height="1440"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-9.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fkinetik-infotainment-9.jpg&w=3840&q=75 1x"
                    style="color: transparent;"
                    width="2400"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="slick-arrow slick-next"
          data-role="none"
          style="display: block;"
          type="button"
        >
           Next
        </button>
      </div>
    </div>
    <div
      class="c-section__slider test"
    >
      <div
        class="c-loading"
      >
        <div
          class="c-loader"
        />
      </div>
      <a
        href="https://volvo-dashboard.netlify.app/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h4>
          Volvo Dashboard
        </h4>
        <p>
          Dashboard system for the Volvo P1800 custom hybrid automobile
        </p>
        <code>
          Technologies used:
          <br />
          Typescript, SCSS, SVG, React, Redux, NodeJS, Express, WebSocket
        </code>
        <svg
          class="c-svg-external-link"
        >
          <use
            xlink:href="#svg-external-link"
          />
        </svg>
      </a>
      <div
        class="slick-slider slick-initialized"
        dir="ltr"
      >
        <button
          class="slick-arrow slick-prev"
          data-role="none"
          style="display: block;"
          type="button"
        >
           Previous
        </button>
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="-2"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Volvo Dashboard screenshot 4"
                    data-nimg="1"
                    decoding="async"
                    height="411"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-4.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-4.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-4.jpg&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1536"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="-1"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Volvo Dashboard screenshot 5"
                    data-nimg="1"
                    decoding="async"
                    height="411"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-5.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-5.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-5.jpg&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1536"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Volvo Dashboard screenshot 1"
                    data-nimg="1"
                    decoding="async"
                    height="411"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-1.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-1.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-1.jpg&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1536"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Volvo Dashboard screenshot 2"
                    data-nimg="1"
                    decoding="async"
                    height="411"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-2.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-2.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-2.jpg&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1536"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="2"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Volvo Dashboard screenshot 3"
                    data-nimg="1"
                    decoding="async"
                    height="411"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-3.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-3.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-3.jpg&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1536"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="3"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Volvo Dashboard screenshot 4"
                    data-nimg="1"
                    decoding="async"
                    height="411"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-4.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-4.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-4.jpg&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1536"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="4"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Volvo Dashboard screenshot 5"
                    data-nimg="1"
                    decoding="async"
                    height="411"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-5.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-5.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-5.jpg&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1536"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="5"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Volvo Dashboard screenshot 1"
                    data-nimg="1"
                    decoding="async"
                    height="411"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-1.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-1.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-1.jpg&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1536"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="6"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Volvo Dashboard screenshot 2"
                    data-nimg="1"
                    decoding="async"
                    height="411"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-2.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-2.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-2.jpg&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1536"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="7"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Volvo Dashboard screenshot 3"
                    data-nimg="1"
                    decoding="async"
                    height="411"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-3.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-3.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-3.jpg&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1536"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="8"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Volvo Dashboard screenshot 4"
                    data-nimg="1"
                    decoding="async"
                    height="411"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-4.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-4.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-4.jpg&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1536"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide slick-cloned"
              data-index="9"
              style="width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                >
                  <img
                    alt="Volvo Dashboard screenshot 5"
                    data-nimg="1"
                    decoding="async"
                    height="411"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-5.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-5.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fautomotive-apps%2Fvolvo-dashboard-5.jpg&w=3840&q=75 2x"
                    style="color: transparent;"
                    width="1536"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="slick-arrow slick-next"
          data-role="none"
          style="display: block;"
          type="button"
        >
           Next
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Unknown component Should unmount the Unknown component 1`] = `<DocumentFragment />`;
